packages/playground/src/bots/GenesisCoinbaseBot.ts

```diff
@@ -0,0 +1,719 @@
+      change: new BigNumber(0.0001),
+      direction: PriceDirection.UP_PERCENTAGE
+    },
+    {
+      token: 'CD10',
+      amount: new BigNumber(1000000000),
+      change: new BigNumber(0.0001),
+      direction: PriceDirection.DOWN_PERCENTAGE
+    },
+    {
+      token: 'TD10',
+      amount: new BigNumber(1000000000),
+      change: new BigNumber(0.0001),
+      direction: PriceDirection.DOWN_PERCENTAGE
+    },
+    {
+      token: 'CS25',
+      amount: new BigNumber(25),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'TS25',
+      amount: new BigNumber(25),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'CR50',
+      amount: new BigNumber(1000),
+      change: new BigNumber(0.33),
+      direction: PriceDirection.RANDOM
+    },
+    {
+      token: 'TR50',
+      amount: new BigNumber(1000),
+      change: new BigNumber(0.33),
+      direction: PriceDirection.RANDOM
+    },
+    {
+      token: 'DFI',
+      amount: new BigNumber(100),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'BTC',
+      amount: new BigNumber(50),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'ETH',
+      amount: new BigNumber(10),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'USDC',
+      amount: new BigNumber(1),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    },
+    {
+      token: 'USDT',
+      amount: new BigNumber(0.99),
+      change: new BigNumber(0),
+      direction: PriceDirection.STABLE
+    }
+  ]
+
+  private async generateToAddress (): Promise<void> {
+    await this.apiClient.call('generatetoaddress', [1, this.OWNER_ADDR, 1], 'number')
+  }
+
+  async fund (amount: number): Promise<void> {
+    await this.apiClient.wallet.sendToAddress(this.OWNER_ADDR, amount)
+    await this.generateToAddress()
+
+    const pubKey = await this.providers.ellipticPair.publicKey()
+    const unspent: any[] = await this.apiClient.wallet.listUnspent(
+      1, 9999999, { addresses: [Bech32.fromPubKey(pubKey, 'bcrt')] }
+    )
+    if (unspent.length === 0) {
+      await this.fund(amount)
+    }
+  }
+
+  async sendTransaction (transaction: TransactionSegWit): Promise<string> {
+    const buffer = new SmartBuffer()
+    new CTransactionSegWit(transaction).toBuffer(buffer)
+    const hex = buffer.toBuffer().toString('hex')
+    return await this.apiClient.rawtx.sendRawTransaction(hex)
+  }
+
+  async bootstrap (): Promise<void> {
+    await this.setup()
+  }
+
+  async cycle (n: number): Promise<void> {
+  }
+
+  async setup (): Promise<void> {
+    await this.setupTokens()
+    await this.setupLoanSchemes()
+    await this.setupOracles()
+    // await new Promise(resolve => setInterval(() => resolve(this.setupOracleData()), 6000))
```

Comment:
no idea how to work like `@interval` in nestjs yet
https://github.com/DeFiCh/playground/blob/d35822dd50955adbffe23f9b66e804a3fa9fb6c5/src/module.playground/bot/oracle.bot.ts#L138

Reply:
move to `cycle`